timely/src/logging.rs

```diff
@@ -142,9 +150,9 @@ pub struct TimelyProgressEvent {
     /// Sequence of nested scope identifiers indicating the path from the root to this instance.
     pub addr: Vec<usize>,
     /// List of message updates, containing Target descriptor, timestamp as string, and delta.
-    pub messages: Box<dyn ProgressEventTimestampVec>,
+    pub messages: std::rc::Rc<dyn ProgressEventTimestampVec>,
```

Comment:
You mentioned that this change was due to needing `Clone` to work out. I think we should probably at least write a note about this explaining to ourselves what is going on here. Another option (h/t @parkmycar)  is to have the traits provide a `box_clone(&self)` method that produces a `Box<dyn Trait>`, though .. we would need to write the `Clone` implementation manually. I'm not 100% sure I grok the fall out with `Rc`, if for example we ever needed to implement `Send`.